### PR TITLE
IntuneAppProtectionPolicyiOS.psm1 issue 1782 - 7 additional parameters added

### LINKED ISSUE #1782: IntuneAppProtectionPolicyiOS Additional Parameters needed to match those available in MSgraph
#### Details of the scenario you tried and the problem that is occurring
pulling back a current configuration for intune, ms graph returns parameters which cannot be configured via Microsoft365DSC 

IsAssigned
ManagedBrowser
MinimumRequiredAppVersion
MinimumRequiredOSVersion
MinimumRequiredSdkVersion 
MinimumWarningAppVersion
MinimumWarningOSVersion

#### Verbose logs showing the problem

[17:20:22][Step 2/4] PowerShell arguments: -NoProfile, -NonInteractive, -ExecutionPolicy, ByPass, -File, D:\BuildAgent\work\1c8ca33856d4ffde\build.ps1
[17:20:23][Step 2/4] At D:\BuildAgent\work\1c8ca33856d4ffde\config\ApplicationProtectionPolicy.ps1:18 char:13
[17:20:23][Step 2/4] +             customBrowserProtocol                   = "http"
[17:20:23][Step 2/4] +             ~~~~~~~~~~~~~~~~~~~~~
[17:20:23][Step 2/4] The member 'customBrowserProtocol' is not valid. Valid members are
[17:20:23][Step 2/4] 'AllowedDataStorageLocations', 'AllowedInboundDataTransferSources', 'AllowedOutboundClipboardSharingLevel', 
[17:20:23][Step 2/4] 'AllowedOutboundDataTransferDestinations', 'AppDataEncryptionType', 'ApplicationId', 'ApplicationSecret', 'Apps', 
[17:20:23][Step 2/4] 'Assignments', 'CertificateThumbprint', 'ContactSyncBlocked', 'Credential', 'DataBackupBlocked', 'DependsOn', 
[17:20:23][Step 2/4] 'Description', 'DeviceComplianceRequired', 'DisplayName', 'Ensure', 'ExcludedGroups', 'FingerprintBlocked', 
[17:20:23][Step 2/4] 'ManagedBrowserToOpenLinksRequired', 'MaximumPinRetries', 'MinimumPinLength', 'OrganizationalCredentialsRequired', 
[17:20:23][Step 2/4] 'PeriodOfflineBeforeAccessCheck', 'PeriodOfflineBeforeWipeIsEnforced', 'PeriodOnlineBeforeAccessCheck', 
[17:20:23][Step 2/4] 'PinCharacterSet', 'PinRequired', 'PrintBlocked', 'PsDscRunAsCredential', 'SaveAsBlocked', 'SimplePinBlocked', 
[17:20:23][Step 2/4] 'TenantId'.
[17:20:23][Step 2/4] At D:\BuildAgent\work\1c8ca33856d4ffde\config\ApplicationProtectionPolicy.ps1:24 char:13
[17:20:23][Step 2/4] +             FaceIsBlocked                           = $False
[17:20:23][Step 2/4] +             ~~~~~~~~~~~~~
[17:20:23][Step 2/4] The member 'FaceIsBlocked' is not valid. Valid members are
[17:20:23][Step 2/4] 'AllowedDataStorageLocations', 'AllowedInboundDataTransferSources', 'AllowedOutboundClipboardSharingLevel', 
[17:20:23][Step 2/4] 'AllowedOutboundDataTransferDestinations', 'AppDataEncryptionType', 'ApplicationId', 'ApplicationSecret', 'Apps', 
[17:20:23][Step 2/4] 'Assignments', 'CertificateThumbprint', 'ContactSyncBlocked', 'Credential', 'DataBackupBlocked', 'DependsOn', 
[17:20:23][Step 2/4] 'Description', 'DeviceComplianceRequired', 'DisplayName', 'Ensure', 'ExcludedGroups', 'FingerprintBlocked', 
[17:20:23][Step 2/4] 'ManagedBrowserToOpenLinksRequired', 'MaximumPinRetries', 'MinimumPinLength', 'OrganizationalCredentialsRequired', 
[17:20:23][Step 2/4] 'PeriodOfflineBeforeAccessCheck', 'PeriodOfflineBeforeWipeIsEnforced', 'PeriodOnlineBeforeAccessCheck', 
[17:20:23][Step 2/4] 'PinCharacterSet', 'PinRequired', 'PrintBlocked', 'PsDscRunAsCredential', 'SaveAsBlocked', 'SimplePinBlocked', 
[17:20:23][Step 2/4] 'TenantId'.
[17:20:23][Step 2/4] At D:\BuildAgent\work\1c8ca33856d4ffde\config\ApplicationProtectionPolicy.ps1:26 char:13
[17:20:23][Step 2/4] +             IsAssigned                              = $True
[17:20:23][Step 2/4] +             ~~~~~~~~~~
[17:20:23][Step 2/4] The member 'IsAssigned' is not valid. Valid members are
[17:20:23][Step 2/4] 'AllowedDataStorageLocations', 'AllowedInboundDataTransferSources', 'AllowedOutboundClipboardSharingLevel', 
[17:20:23][Step 2/4] 'AllowedOutboundDataTransferDestinations', 'AppDataEncryptionType', 'ApplicationId', 'ApplicationSecret', 'Apps', 
[17:20:23][Step 2/4] 'Assignments', 'CertificateThumbprint', 'ContactSyncBlocked', 'Credential', 'DataBackupBlocked', 'DependsOn', 
[17:20:23][Step 2/4] 'Description', 'DeviceComplianceRequired', 'DisplayName', 'Ensure', 'ExcludedGroups', 'FingerprintBlocked', 
[17:20:23][Step 2/4] 'ManagedBrowserToOpenLinksRequired', 'MaximumPinRetries', 'MinimumPinLength', 'OrganizationalCredentialsRequired', 
[17:20:23][Step 2/4] 'PeriodOfflineBeforeAccessCheck', 'PeriodOfflineBeforeWipeIsEnforced', 'PeriodOnlineBeforeAccessCheck', 
[17:20:23][Step 2/4] 'PinCharacterSet', 'PinRequired', 'PrintBlocked', 'PsDscRunAsCredential', 'SaveAsBlocked', 'SimplePinBlocked', 
[17:20:23][Step 2/4] 'TenantId'.
[17:20:23][Step 2/4] At D:\BuildAgent\work\1c8ca33856d4ffde\config\ApplicationProtectionPolicy.ps1:32 char:13
[17:20:23][Step 2/4] +             MinimumRequiredOSVersion                = "13.0.0"
[17:20:23][Step 2/4] +             ~~~~~~~~~~~~~~~~~~~~~~~~
[17:20:23][Step 2/4] The member 'MinimumRequiredOSVersion' is not valid. Valid members are
[17:20:23][Step 2/4] 'AllowedDataStorageLocations', 'AllowedInboundDataTransferSources', 'AllowedOutboundClipboardSharingLevel', 
[17:20:23][Step 2/4] 'AllowedOutboundDataTransferDestinations', 'AppDataEncryptionType', 'ApplicationId', 'ApplicationSecret', 'Apps', 
[17:20:23][Step 2/4] 'Assignments', 'CertificateThumbprint', 'ContactSyncBlocked', 'Credential', 'DataBackupBlocked', 'DependsOn', 
[17:20:23][Step 2/4] 'Description', 'DeviceComplianceRequired', 'DisplayName', 'Ensure', 'ExcludedGroups', 'FingerprintBlocked', 
[17:20:23][Step 2/4] 'ManagedBrowserToOpenLinksRequired', 'MaximumPinRetries', 'MinimumPinLength', 'OrganizationalCredentialsRequired', 
[17:20:23][Step 2/4] 'PeriodOfflineBeforeAccessCheck', 'PeriodOfflineBeforeWipeIsEnforced', 'PeriodOnlineBeforeAccessCheck', 
[17:20:23][Step 2/4] 'PinCharacterSet', 'PinRequired', 'PrintBlocked', 'PsDscRunAsCredential', 'SaveAsBlocked', 'SimplePinBlocked', 
[17:20:23][Step 2/4] 'TenantId'.
[17:20:23][Step 2/4] At D:\BuildAgent\work\1c8ca33856d4ffde\config\ApplicationProtectionPolicy.ps1:37 char:13
[17:20:23][Step 2/4] +             PeriodBeforePinReset                    = "P60D"
[17:20:23][Step 2/4] +             ~~~~~~~~~~~~~~~~~~~~
[17:20:23][Step 2/4] The member 'PeriodBeforePinReset' is not valid. Valid members are
[17:20:23][Step 2/4] 'AllowedDataStorageLocations', 'AllowedInboundDataTransferSources', 'AllowedOutboundClipboardSharingLevel', 
[17:20:23][Step 2/4] 'AllowedOutboundDataTransferDestinations', 'AppDataEncryptionType', 'ApplicationId', 'ApplicationSecret', 'Apps', 
[17:20:23][Step 2/4] 'Assignments', 'CertificateThumbprint', 'ContactSyncBlocked', 'Credential', 'DataBackupBlocked', 'DependsOn', 
[17:20:23][Step 2/4] 'Description', 'DeviceComplianceRequired', 'DisplayName', 'Ensure', 'ExcludedGroups', 'FingerprintBlocked', 
[17:20:23][Step 2/4] 'ManagedBrowserToOpenLinksRequired', 'MaximumPinRetries', 'MinimumPinLength', 'OrganizationalCredentialsRequired', 
[17:20:23][Step 2/4] 'PeriodOfflineBeforeAccessCheck', 'PeriodOfflineBeforeWipeIsEnforced', 'PeriodOnlineBeforeAccessCheck', 
[17:20:23][Step 2/4] 'PinCharacterSet', 'PinRequired', 'PrintBlocked', 'PsDscRunAsCredential', 'SaveAsBlocked', 'SimplePinBlocked', 
[17:20:23][Step 2/4] 'TenantId'.
[17:20:23][Step 2/4] At D:\BuildAgent\work\1c8ca33856d4ffde\config\ApplicationProtectionPolicy.ps1:49 char:9
[17:20:23][Step 2/4] +         IntuneAppProtectionPolicyAndroid 'Android_MAM_General'
[17:20:23][Step 2/4] +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[17:20:23][Step 2/4] Undefined DSC resource 'IntuneAppProtectionPolicyAndroid'. Use Import-DSCResource to import the resource.
[17:20:23][Step 2/4]     + CategoryInfo          : ParserError: (:) [], ParseException
[17:20:23][Step 2/4]     + FullyQualifiedErrorId : InvalidInstanceProperty

#### Suggested solution to the issue

Add these prameters to the resource

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
IntuneAppProtectionPolicyiOS 'iOS_MAM_General'
        {
            AllowedDataStorageLocations             = @("oneDriveForBusiness","sharePoint")
            AllowedInboundDataTransferSources       = "managedApps"
            AllowedOutboundClipboardSharingLevel    = "managedApps"
            AllowedOutboundDataTransferDestinations = "managedApps"
            AppDataEncryptionType                   = "whenDeviceLocked"
            ApplicationId                           = $ConfigurationData.NonNodeData.ApplicationId;
            ApplicationSecret                       = $ConfigurationData.NonNodeData.ApplicationSecret;
            #Apps                                    = @("com.my.appsHere") #Need to list out the app com addresses
            #Assignments                             = @("ms-Pilot-SG") #Needs to link to the AD group
            #CertificateThumbprint                   = ""
            ContactSyncBlocked                      = $False
            customBrowserProtocol                   = "http"
            DataBackupBlocked                       = $True
            Description                             = "Microsoft Application Management policy for general 365 iOS apps"
            DeviceComplianceRequired                = $True
            DisplayName                             = "iOS MAM General"
            #ExcludedGroups                          = @("NOT_SET")
            FaceIDBlocked                           = $False
            FingerprintBlocked                      = $False
            IsAssigned                              = $True
            ManagedBrowser                          = ""
            ManagedBrowserToOpenLinksRequired       = $True
            MaximumPinRetries                       = 5
            MinimumPinLength                        = 8
            MinimumRequiredAppVersion               = "1"
            MinimumRequiredOSVersion                = "13.0.0"
            MinimumRequiredSdkVersion               = "0.1"
            MinimumWarningAppVersion                = "0.1"
            MinimumWarningOSVersion                 = "13.0.0"
            OrganizationalCredentialsRequired       = $False
            PeriodBeforePinReset                    = "P60D"
            PeriodOfflineBeforeAccessCheck          = "P1D"
            PeriodOfflineBeforeWipeIsEnforced       = "P60D"
            PeriodOnlineBeforeAccessCheck           = "PT15M"
            PinCharacterSet                         = "alphanumericAndSymbol"
            PinRequired                             = $True
            PrintBlocked                            = $True
            SaveAsBlocked                           = $True
            SimplePinBlocked                        = $True
            Ensure                                  = 'Present'
            Credential                              = $credsGlobalAdmin
            TenantId                                 = $ConfigurationData.NonNodeData.TenantId;
            #PsDscRunAsCredential                + 
        }
```

#### Version of the DSC module that was used ('dev' if using current dev branch)
Dev

### PR DESCRIPTION
**Pull request by Jamiet19**
#### Pull Request (PR) description
in this change I have added 7 addiotional parameters which are supported by MS graph but did not feature in the module.
The unit test has been updated to include the additional parameters as they are all required fields.

#### This Pull Request (PR) fixes the following issues

    - Fixes #1782

### CODE REVIEW

**Comment by Jamiet19** [comment]
I missed the schema.mof update with this pull request. added to PR

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneAppProtectionPolicyiOS/MSFT_IntuneAppProtectionPolicyiOS.schema.mof [hunk 1]
```diff
@@ -25,6 +25,13 @@ class MSFT_IntuneAppProtectionPolicyiOS : OMI_BaseResource
     [Write, Description("N/A")] Boolean PrintBlocked;
     [Write, Description("N/A")] Boolean FingerprintBlocked;
     [Write, Description("N/A")] Boolean FaceIdBlocked;
+    [Write, Description("N/A")] Boolean IsAssigned;
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneAppProtectionPolicyiOS/MSFT_IntuneAppProtectionPolicyiOS.schema.mof:28]: I am aware that this is the same for the already existing parameters, but do you have a description for these new parameter (and if possible also the already existing parameters 😉)? That way the generated documentation will be more useful.

**Review by ykuijs — CHANGES_REQUESTED** [review]
Could you also update the Changelog to reflect this change?

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneAppProtectionPolicyiOS/MSFT_IntuneAppProtectionPolicyiOS.schema.mof (hunk 1 above)
Comment (NikCharlebois) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneAppProtectionPolicyiOS/MSFT_IntuneAppProtectionPolicyiOS.schema.mof:28]: @Jamiet19 can you look at providing a description as highlighted by Yorick? We will approve the PR right after. Thanks

Comment (Jamiet19) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneAppProtectionPolicyiOS/MSFT_IntuneAppProtectionPolicyiOS.schema.mof]: > @Jamiet19 can you look at providing a description as highlighted by Yorick? We will approve the PR right after. Thanks

I've updated the descriptions for all parameters that didn't already have them. I've also added another parameter I discovered was missing.

**Review by NikCharlebois — APPROVED** [review]

**Comment by NikCharlebois** [comment]
@ykuijs can you finalize the review when you get a chance

**Review by ykuijs — CHANGES_REQUESTED** [review]
Could you please add your changes to the changelog

**Comment by Jamiet19** [comment]
> Could you please add your changes to the changelog

HI @ykuijs Apologies, I'm unsure how version numbers work in GitHub. Is there any guidance I can refer to to make sure I use an appropriate version number - unless you're able to suggest one in line with the updates I've made?

Thanks.

**Comment by ykuijs** [comment]
You can add your changed in the Unreleased section of the [changelog](https://github.com/microsoft/Microsoft365DSC/blob/Dev/CHANGELOG.md), like it is done currently as well:
```
# UNRELEASED

* EXOCASMailboxPlan
  * Add support for DisplayName as identifier for CAS mailbox plan.
* EXOTransportConfig
  * New resource for Exchange Online transport configuration.
* DEPENDENCIES
  * Updated DSCParser to 1.3.0.4.
  * Updated Microsoft.Graph.* to 1.9.3.
```
 
You can just add something like (try to do this in alphabetical order 😉):
* IntuneAppProtectionPolicyiOS
  * Add 7 additional parameters to the resource and added parameter descriptions

**Comment by NikCharlebois** [comment]
Hey @Jamiet19 Let me handle the Change log on my end this time. I am working on today's release. After it's done, just have a look at the Changelog.md file for future reference. Thanks for all your help!

**Comment by Jamiet19** [comment]
> Hey @Jamiet19 Let me handle the Change log on my end this time. I am working on today's release. After it's done, just have a look at the Changelog.md file for future reference. Thanks for all your help!

Thanks Nik - Understood for next time.

**Pull request merged**